packages/aws-cdk/lib/api/cloudformation-deployments.ts

```diff
@@ -366,6 +360,115 @@ export class CloudFormationDeployments {
     return stack.exists;
   }
 
+  private async readCurrentNestedTemplate(stackArtifact: cxapi.CloudFormationStackArtifact, nestedStackName: string): Promise<Template> {
+    return this.readCurrentStackTemplate(stackArtifact, nestedStackName);
+  }
+
+  private async replaceNestedStacksInRootStack(rootStackArtifact: cxapi.CloudFormationStackArtifact, deployedTemplate: any) {
+    if (!this.stackHasNestedStacks(rootStackArtifact)) {
+      return;
+    }
+
+    const sdkWithRole = await prepareSdkWithLookupRoleFor(this.sdkProvider, rootStackArtifact);
+    const listStackResourcesByStackName: { [key:string] : ListStackResources } = {};
+    await this.replaceNestedStacksInParentTemplate(
+      rootStackArtifact, rootStackArtifact.template, deployedTemplate, rootStackArtifact.stackName, listStackResourcesByStackName, sdkWithRole.sdk,
+    );
+  }
+
+  private stackHasNestedStacks(stackArtifact: cxapi.CloudFormationStackArtifact): boolean {
+    return Object.values(stackArtifact.template.Resources ?? {}).some((resource: any) => resource.Type === 'AWS::CloudFormation::Stack');
+  }
+
+  private async replaceNestedStacksInParentTemplate(
+    rootStackArtifact: cxapi.CloudFormationStackArtifact, generatedParentTemplate: any, deployedParentTemplate: any,
+    parentStackName: string | undefined, listStackResourcesByStackName: { [key: string] : ListStackResources }, sdk: ISDK,
+  ) {
+    const idResourceMap = Object.entries(generatedParentTemplate.Resources ?? {}).filter(([_, attrs]) => this.isNestedStackResource(attrs));
+    for (const [nestedStackLogicalId, resource] of idResourceMap) {
+      let nestedStackResource = resource as any;
+
+      if (parentStackName && !listStackResourcesByStackName[parentStackName]) {
+        listStackResourcesByStackName[parentStackName] = new LazyListStackResources(sdk, parentStackName);
+      }
```

Comment:
Can you explain why do we need this `listStackResourcesByStackName` map? Why can't we just create a new `LazyListStackResources` in this function, before the `for` loop, and just use that? We never call this function with the same Stack twice anyway, right?